src/google/protobuf/compiler/cpp/cpp_file.cc

```diff
@@ -824,19 +824,11 @@ void FileGenerator::GenerateNamespaceClosers(io::Printer* printer) {
 
 void FileGenerator::GenerateForwardDeclarations(io::Printer* printer) {
```

Comment:
It is called from FileGenerator::GenerateHeader().
That call is still neccesary for the types that are declared in the main protobuf file.
Only the dependencies are unneccesary. This is consistent with the generated .pb.cc file where the *\_default\_instance\_ variables are not written for dependencies either.